src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -759,8 +849,15 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                     {
                         context.LastDatabaseChangeVector = updatedChangeVector.ChangeVector;
                     }
+
+                    lastIndexInBatch = long.Max(lastIndexInBatch, command.Index);
                 }
 
+                // set last cluster transaction index (persistent)
+                var lastClusterTxIndex = DocumentsStorage.ReadLastCompletedClusterTransactionIndex(context.Transaction.InnerTransaction);
+                if(lastIndexInBatch > lastClusterTxIndex)
+                    Database.DocumentsStorage.SetLastCompletedClusterTransactionIndex(context, lastIndexInBatch);
```

Comment:
Shouldn't we set DocumentDatabase.LastCompletedClusterTransactionIndex to this value when tx is commited? If not there is a time gap in theory huge) between commiting and reading that in the thread. What issues can it cause?

Reply:
`Database.DocumentsStorage.SetLastCompletedClusterTransactionIndex` sets a persistent variable, so it will be set only on the tx commit.
This promises that the value will be updated only when the cluster transaction has been committed.

Reply:
Check `AfterCommitWhenNewTransactionsPrevented` event

Reply:
After discussing this with Karmel, it is not required here..